### PR TITLE
make onnx_decompose_transpose test conditional on ONNX_MLIR_DECOMP_ON…

### PR DESCRIPTION
**Pull request by mgehre-amd**
…NX_CONVTRANSPOSE

Since introduction of the ONNX_MLIR_DECOMP_ONNX_CONVTRANSPOSE option, the onnx_decompose_transpose.mlir test would fail whenever it was set to OFF.

Now ensure that we only run this test when the variable is ON.

This is required to make the bump automation work (when trying to check if tests pass).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -210,6 +210,9 @@ endif()
 
 if (ONNX_MLIR_DECOMP_ONNX_CONVTRANSPOSE)
   add_compile_definitions(ONNX_MLIR_DECOMP_ONNX_CONVTRANSPOSE)
+  set(ONNX_MLIR_DECOMP_ONNX_CONVTRANSPOSE_ENABLED 1)
```
Comment (TinaAMD) [inline CMakeLists.txt:213]: Why can't you reuse `ONNX_MLIR_DECOMP_ONNX_CONVTRANSPOSE`?

Comment (mgehre-amd) [reply CMakeLists.txt:213]: ONNX_MLIR_DECOMP_ONNX_CONVTRANSPOSE can be ON, OFF, YES, NO, 0, 1, but we need 0 or 1 when substituting into lit

**Review by TinaAMD — APPROVED** [review]
LGTM

**Review by cferry-AMD — APPROVED** [review]
LGTM

**Pull request merged**